.github/workflows/pr-validation.yaml

```diff
@@ -0,0 +1,62 @@
+name: PR validity
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, edited]
+
+jobs:
+  check_testing_header:
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Check for `### Testing` section in PR
+        id: check-testing
+        continue-on-error: true
+        run: |
+          python -c 'import sys; pr_summary = """${{ github.event.pull_request.body }}"""; sys.exit(0 if "### Testing" in pr_summary else 1)'
```

Comment:
This breaks for any PR that has single quotes.  That's not viable.  Those are not used solely in contractions, and most importantly _they can be, and are, used in logs_.

Reply:
Noticed this after the fact, will fix it as a separate PR (or if urgent we can roll this back and then put it back in)